src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -313,6 +313,74 @@ public String toString() {
      **/
     private static final Lock sLock = new ReentrantLock();
 
+    static final String USER_AGENT_TEMPLATE = "%s|%s|%s|%s|%s|%s|%s";
+    static final String userAgentStr;
+
+    static {
+        userAgentStr = getUserAgent();
+    }
+
+    static String getUserAgent() {
+        try {
+            return String.format(
+                    USER_AGENT_TEMPLATE,
+                    "1",
+                    "MS-JDBC",
+                    getJDBCVersion(),
+                    getOSType(),
+                    getOSDetails(),
+                    getArchitecture(),
+                    getRuntimeDetails()
+                    );
+        } catch(Exception e) {
+            return "MS-JDBC";
```

Comment:
A better fallback would probably include the version: "1|MS-JDBC"

I'd also eliminate the string duplication:
```
static final String USER_AGENT_TEMPLATE = "%s|%s|%s|%s|%s|%s";
static final String EXT_VERSION_AND_DRIVER_NAME = "1|MS-JDBC";
...
            return String.format(
                    USER_AGENT_TEMPLATE,
                    EXT_VERSION_AND_DRIVER_NAME,
                    getJDBCVersion(),
...
        } catch(Exception e) {
            return EXT_VERSION_AND_DRIVER_NAME;
...